### PR TITLE
feat(web): reset-password

### PR DESCRIPTION
**Pull request by Loris-Reggiani**
# Description

Adding on the login page an option to reset the password using the route reset (put)
Adding reset page witch can be opened by the mail received after demanding it  on the login page, this page use route reset (post)

# Fixes

Fixed esLint

# Changes include

- [ADD] reset page
- [ADD] option in login page to reset password
# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added @vrn-sh/members to the reviewers, or specific members of the team
- [ ] I have added the needed labels
- [ ] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: web/src/pages/Login/Login.tsx [hunk 1]
```diff
@@ -158,7 +147,39 @@ export default function Login() {
...
+                                }}
+                            >
+                                <h2>Reset Password</h2>
+                                <p>Please enter your email address to reset your password.</p>
+                                <input
+                                    type="email"
+                                    value={resetEmail}
+                                    onChange={(e) => setResetEmail(e.target.value)}
+                                    placeholder="Enter your email"
+                                    // Add any additional input styles
```
Comment (djnnvx) [inline web/src/pages/Login/Login.tsx]: what is this comment doing here ?

File: web/src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -158,7 +147,39 @@ export default function Login() {
...
+                                />
+                                <button onClick={handlePasswordReset}>Send Email</button>
+                                <p>{resetFeedback}</p> {/* Feedback message */}
```
Comment (djnnvx) [inline web/src/pages/Login/Login.tsx]: you cannot just put `/* Feedback message */` here, you need to get the error message from the back-end and display it

File: web/src/pages/Login/Login.tsx [hunk 3]
```diff
@@ -87,36 +96,16 @@ export default function Login() {
                         Cookies.set('Id', e.data.id, {
                             expires: Date.parse(e.data.expiry),
                         });
-                        Cookies.set('Fav', '', {
```
Comment (djnnvx) [inline web/src/pages/Login/Login.tsx]: why did you remove this

File: web/src/pages/Login/Login.tsx [hunk 4]
```diff
@@ -76,8 +86,7 @@ export default function Login() {
                         }
                     )
                     .then((e) => {
-                        setMessage('Connecting...', 'success');
-                        navigate('/accueil');
```
Comment (djnnvx) [inline web/src/pages/Login/Login.tsx]: why change this back ? Clara did a good job with the refactoring of the main page, please don't change it

**Review by djnnvx — CHANGES_REQUESTED** [review]
this is not a good PR:

1. You're missing the page that calls `/reset-password` after the user has received the email. it should ask the user for their new password & call the view with the token received in the mail
2. you're removing or changing code unrelated from your issue, but this will break other people's issues. you need to make sure you don't break things for others
3. lot of unnecessary comments. this is not a big deal, but just a bit weird. did you generate this with chatgpt ? 


@claraxuxu can you check whether the changes in the front-end are harmful please ? i don't know why the key handler has been removed for instance

File: web/src/pages/Login/Login.tsx (hunk 4 above)
Comment (Loris-Reggiani) [reply web/src/pages/Login/Login.tsx]: sorry, I didn't see I deleted some elements of Login(), it should be fix now

File: web/src/pages/Login/Login.tsx [hunk 5]
```diff
@@ -28,6 +29,29 @@ export default function Login() {
...
     };
 
+    const [isResetModalOpen, setIsResetModalOpen] = useState(false);
+    const [resetEmail, setResetEmail] = useState('');
+    const [resetFeedback, setResetFeedback] = useState('');
+    const [isSuccess, setIsSuccess] = useState(false);
+
+    const handlePasswordReset = async () => {
+        try {
+            const response = await axios.put(`${config.apiUrl}reset`, {
```
Comment (claraxuxu) [inline web/src/pages/Login/Login.tsx]: manque '/' dans l'url

**Comment by claraxuxu** [comment]
- [ ] When mouse hover on the button, the cursor is not pointer
- [ ] Size of Modal is too large
- [ ] Two buttons in Modal should have different level, one primary one secondary
- [ ] Input not in size
- [ ] Warning text better in one line

截屏2023-11-17 10 43 58

File: web/src/pages/Login/Login.tsx (hunk 5 above)
Comment (claraxuxu) [reply web/src/pages/Login/Login.tsx]: @Loris-Reggiani manque toujours de '/'

Comment (Loris-Reggiani) [reply web/src/pages/Login/Login.tsx]: > @Loris-Reggiani manque toujours de '/'

ok, c'est fix (encore désolé pour ce /)

File: web/src/pages/Dashboard/Accueil.tsx
Comment (claraxuxu) [inline web/src/pages/Dashboard/Accueil.tsx:1]: supprimer ces changements stp

File: web/src/pages/Login/Login.tsx [hunk 6]
```diff
@@ -158,7 +182,98 @@ export default function Login() {
...
+                                >
+                                    <span className="txt-color">
+                                        reset my password
+                                    </span>
+                                </div>
+                                <Modal
+                                    isOpen={isResetModalOpen}
+                                    onRequestClose={toggleResetModal}
+                                    contentLabel="Reset Password Modal"
+                                    style={{
```
Comment (claraxuxu) [inline web/src/pages/Login/Login.tsx]: à mettre dans le fichier scss

File: web/src/pages/Reset/Reset.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+import '../Login/Login.scss';
+
+export default function Reset() {
+    const [resetPassword, setResetPassword] = useState('');
+    const [resetFeedback, setResetFeedback] = useState('');
+    const [isSuccess, setIsSuccess] = useState(false);
+    const urlParams = new URLSearchParams(window.location.search);
+    const token = urlParams.get('token');
+
+    console.log(token);
```
Comment (claraxuxu) [inline web/src/pages/Reset/Reset.tsx]: à supprimer, ça va cause des pbl sécu

**Review by claraxuxu — COMMENTED** [review]
niveau front simple me va, faut tester l'api

**Review by claraxuxu — APPROVED** [review]
Front end checked
